service/storage_service.cc

```diff
@@ -863,7 +863,7 @@ future<> storage_service::check_and_repair_cdc_streams() {
         }
         const auto new_streams_ts = cdc::make_new_cdc_generation(db().local().get_config(),
                 {}, std::move(tmptr), _gossiper,
-                _sys_dist_ks.local(), get_ring_delay(), false /* for_testing */);
+                _sys_dist_ks.local(), get_ring_delay(), !is_first_node() /* add delay */);
```

Comment:
When we create a new generation in an existing cluster it must have a delay! At this point it doesn't matter which node it is.

The trick of not adding a delay on the first node works because it's the **only** node, so the generation doesn't need to be gossiped around the cluster.

Reply:
The PR letter should also explain why the delay was added in the first place and why it's safe not to add the delay for the first node.

Reply:
Right. I confussed is_first with is_only

Reply:
Fixed.